server/controllers/placementApplications/reviewController.test.ts

```diff
@@ -0,0 +1,358 @@
+      ;(fetchErrorsAndUserInput as jest.Mock).mockReturnValue(errorsAndUserInput)
+
+      const placementApplication = placementApplicationFactory.build()
+      placementApplicationService.getPlacementApplication.mockResolvedValue(placementApplication)
+      ;(PlacementApplicationReview as jest.Mock).mockImplementation(() => {
+        return { applicationId: id, step: 'review' }
+      })
+
+      const requestHandler = reviewController.show('review')
+
+      await requestHandler(request, response, next)
+
+      expect(response.render).toHaveBeenCalledWith('placement-applications/pages/review/review', {
+        pageProps: {
+          pageHeading: 'Review information',
+          backLink: `${matchPaths.placementRequests.index}#placement-applications`,
+        },
+        placementApplication,
+        errors: errorsAndUserInput.errors,
+        errorSummary: errorsAndUserInput.errorSummary,
+        ...errorsAndUserInput.userInput,
+      })
+    })
+
+    it('shows errors and user input on decision page when returning from an error state', async () => {
+      const errorsAndUserInput = createMock<ErrorsAndUserInput>()
+      ;(fetchErrorsAndUserInput as jest.Mock).mockReturnValue(errorsAndUserInput)
+
+      const placementApplication = placementApplicationFactory.build()
+      placementApplicationService.getPlacementApplication.mockResolvedValue(placementApplication)
+      ;(PlacementApplicationReview as jest.Mock).mockImplementation(() => {
+        return { applicationId: id, step: 'decision' }
+      })
+
+      const requestHandler = reviewController.show('decision')
+
+      await requestHandler(request, response, next)
+
+      expect(response.render).toHaveBeenCalledWith('placement-applications/pages/review/decision', {
+        pageProps: {
+          pageHeading: 'Make a decision',
+          backLink: placementApplicationPaths.placementApplications.show({ id }),
+        },
+        placementApplication,
+        errors: errorsAndUserInput.errors,
+        errorSummary: errorsAndUserInput.errorSummary,
+        ...errorsAndUserInput.userInput,
+      })
+    })
+  })
+
+  describe('update', () => {
+    const request: DeepMocked<Request> = createMock<Request>({
+      headers: { referer: 'some-referrer' },
+      params: { id },
+      flash: jest.fn(),
+      session: createMock<Request['session']>({
+        placementApplicationDecisions: {},
+      }),
+    })
+
+    it('redirects to the decision path on review update success', async () => {
+      ;(PlacementApplicationReview as jest.Mock).mockImplementation(() => {
+        return { applicationId: id, update: jest.fn() }
+      })
+
+      const requestHandler = reviewController.update()
+
+      await requestHandler(
+        {
+          ...request,
+          body: {
+            summaryOfChanges: 'some changes',
+          },
+        },
+        response,
+      )
+
+      expect(response.redirect).toHaveBeenCalledWith(
```

Comment:
it might be a bit fiddly but I think we should be asserting that `PlacementApplicationReview.update` has been called here too.